warehouse/templates/index.html

```diff
@@ -84,35 +97,39 @@ <h1 class="homepage-banner__title">Find, install and publish Python packages wit
   </div>
 </section>
 
-<section class="horizontal-section horizontal-section--grey">
-  <div class="site-container">
+<section class="horizontal-section horizontal-section--grey" data-controller="increment">
+  <div class="site-container" >
     <div class="col-half">
       <div class="heading-wsubtitle">
         <h2 class="heading-wsubtitle__heading">Trending Projects</h2>
         <p class="heading-wsubtitle__subtitle">Trending projects as downloaded by the community</p>
       </div>
-      <div class="package-list">
+      <div class="package-list-timeout">
+        {% set current = namespace(amount_to_show=0) %}
         {% for release in trending_projects %}
-          {{ project_snippet(release) }}
+          {% if current.amount_to_show < trending_projects_visible.total %}
+            {{ project_snippet_trending(release, release.summary, current.amount_to_show) }}
+            {% set current.amount_to_show = current.amount_to_show + 1 %}
+          {% endif %}
```

Comment:
this loop might be clearer simply iterating over an index? something like

```
{% for i in range([trending_projects_visible.total, trending_projects|length]|min) %}
  {{ project_snippet_trending(trending_projects[i], trending_projects[i].summary, i) }}
{% endfor %}
```

(the range argument is admittedly awkward due to [`min`](http://jinja.pocoo.org/docs/dev/templates/#min) and [`length`](http://jinja.pocoo.org/docs/dev/templates/#length) being Jinja2 template filters)

Reply:
made it much more clear now, actually how it was originally, I did not need all those extra jinga variables